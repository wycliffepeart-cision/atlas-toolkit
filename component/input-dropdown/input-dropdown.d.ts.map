{"version":3,"file":"input-dropdown.d.ts","sourceRoot":"","sources":["../../../src/component/input-dropdown/input-dropdown.ts"],"names":[],"mappings":"AAKA,qBACa,aAAc,SAAQ,WAAW;IAC5C,MAAM,KAAK,kBAAkB,aAE5B;CACF"}