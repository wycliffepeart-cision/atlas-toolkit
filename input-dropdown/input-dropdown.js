import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var InputDropdownTemplate = "<template>\n  <slot name=\"input\" part=\"input\"></slot>\n  <slot name=\"menu\" part=\"menu\"></slot>\n</template>\n";

const DROPDOWN_ATTRIBUTES = ['open'];
let InputDropdown = class InputDropdown extends HTMLElement {
    static get observedAttributes() {
        return DROPDOWN_ATTRIBUTES;
    }
};
InputDropdown = __decorate([
    Template(InputDropdownTemplate)
], InputDropdown);

export { InputDropdown };
