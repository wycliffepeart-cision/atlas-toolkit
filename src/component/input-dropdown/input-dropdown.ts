import { Template } from '../../decorators/template';
import InputDropdownTemplate from './input-dropdown.html';

const DROPDOWN_ATTRIBUTES = ['open'];

@Template(InputDropdownTemplate)
export class InputDropdown extends HTMLElement {
  static get observedAttributes() {
    return DROPDOWN_ATTRIBUTES;
  }
}
