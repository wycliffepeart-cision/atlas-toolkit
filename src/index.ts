export * from './component/accordion/accordion';
export * from './component/alert/alert';
export * from './component/application-frame/application-frame';
export * from './component/button/button';
export * from './component/icon/icon';
export * from './component/button-set/button-set';
export * from './component/chip/chip';
export * from './component/chip-group/chip-group';
export * from './component/dialog/dialog';
export * from './component/feed/feed';
export * from './component/input-checkbox/input-checkbox';
export * from './component/input-chip-group/input-chip-group';
export * from './component/input-dropdown/input-dropdown';
export * from './component/input-search/input-search';
export * from './component/input-text/input-text';
export * from './component/input-textarea/input-textarea';
export * from './component/logo/logo';
export * from './component/main-nav/main-nav';
export * from './component/main-nav-item/main-nav-item';
export * from './component/number-display/number-display';
export * from './component/profile-control/profile-control';
export * from './component/snackbar/snackbar';
export * from './component/stepper/stepper';
export * from './component/stepper-step/stepper-step';
export * from './component/tile-board/tile-board';
export * from './component/tile-board-tile/tile-board-tile';
export * from './component/workflow/workflow';
export * from './component/workflow-step/workflow-step';
