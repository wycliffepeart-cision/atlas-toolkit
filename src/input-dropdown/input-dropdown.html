<template>
	<slot name="input" part="input"></slot>
	<slot name="menu" part="menu"></slot>
</template>
