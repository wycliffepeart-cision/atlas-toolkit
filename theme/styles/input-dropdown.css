@layer atlas {
	atlas-dropdown {
		--default-width: 100%;
		--menu-padding: 1em;
		--menu-margin-block-start: 0.35em;
		--menu-bg-color: var(--atlas-input-common-background-color);
	}

	atlas-dropdown {
		position: relative;
		isolation: isolate;
		display: inline-block;
		width: var(--default-width);
	}

	atlas-dropdown::part(menu) {
		display: none;

		max-height: fit-content;
		width: calc(var(--default-width) - (var(--menu-padding) * 2));
		padding: var(--menu-padding);
		margin-block-start: var(--menu-margin-block-start);

		background-color: var(--menu-bg-color);
		position: absolute;
		overflow-y: auto;

		border: none;
		box-shadow: var(--atlas-input-common-box-shadow);
		border-radius: var(--atlas-input-common-border-radius);
	}

	atlas-dropdown[open]:not([open="false"])::part(menu) {
		display: block;
	}
}
