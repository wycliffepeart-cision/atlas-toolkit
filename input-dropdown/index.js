import { InputDropdown } from './input-dropdown.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-dropdown'))
    customElements.define('atlas-dropdown', InputDropdown);
