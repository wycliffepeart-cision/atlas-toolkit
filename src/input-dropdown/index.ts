import { InputDropdown } from './input-dropdown';

if (!customElements.get('atlas-dropdown')) customElements.define('atlas-dropdown', InputDropdown);
