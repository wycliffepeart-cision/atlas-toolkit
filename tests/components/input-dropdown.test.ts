import '../../src/input-dropdown';

const ELEMENT_TAG = 'atlas-dropdown';
const getShadowDomContent = (element) => element.shadowRoot.innerHTML;

afterEach(() => {
  const elem = document.querySelector('body > *:first-child');
  elem?.remove();

  jest.clearAllMocks();
});

describe('Dropdown', () => {
  it('exists', async () => {
    await customElements.whenDefined(ELEMENT_TAG);
    const element = document.createElement(ELEMENT_TAG);
    document.body.appendChild(element);

    (expect(element) as any).toBeInTheDocument();
  });

  it('is matches snapshot', () => {
    const element = document.createElement(ELEMENT_TAG);
    document.body.appendChild(element);

    (expect(getShadowDomContent(element)) as any).toMatchSnapshot();
  });
	
});
