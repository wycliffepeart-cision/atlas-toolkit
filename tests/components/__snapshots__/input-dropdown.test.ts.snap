// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown is matches snapshot 1`] = `
<slot name="input"
      part="input"
>
</slot>
<slot name="menu"
      part="menu"
>
</slot>
`;
